Update README.md to specify gathered telemetry (#16379)
PowerShell/PowerShell: README.md

### Telemetry

Sending telemetry is controlled via an environment variable called `POWERSHELL_TELEMETRY_OPTOUT`.
Set the value to `1`, `yes`, or `true` before starting PowerShell to disable telemetry.

> Setting the value to `0`, `no`, `false`, or not creating the variable will enable telemetry.

Reviewer: Isn't it more accurate that if it's anything other than `1`,`yes`, or `true` then it defaults to false meaning to send telemetry?

The telemetry we collect falls under the [Microsoft Privacy Statement](https://privacy.microsoft.com/privacystatement/).

The following is sent when the shell starts:

> - Operating system information

Reviewer: it would be great to make these links to the source code where we collect this information

Reviewer 2: Make sure to use the [permanent links to files](https://docs.github.com/en/repositories/working-with-files/using-files/getting-permanent-links-to-files).

Author: @SteveL-MSFT - this could be problematic if the code changes

- The commit id of the build
- The enabled experimental features
- A randomly generated guid representing the session
- A randomly generated guid representing the system
- Geo-location information is supplied by Application Insights (Country, Region, State, City, if available)
- The value of the environment variable `POWERSHELL_DISTRIBUTION_CHANNEL`

During shell operation the following is sent periodically:

> - The name of a module when loaded if it is in the list of modules (see line 128 in src/System.Management.Automation/utils/Telemetry.cs).

Reviewer: make that a link instead of text to the source line

Reviewer 2: Make sure to use the [permanent links to files](https://docs.github.com/en/repositories/working-with-files/using-files/getting-permanent-links-to-files).

> - The name of a module when loaded if it is in the list of modules (see line 128 in src/System.Management.Automation/utils/Telemetry.cs).

Reviewer: ```suggestion
- The name of a module when loaded if it is in the pre-approved list of modules (see line 128 in src/System.Management.Automation/utils/Telemetry.cs).
```

> - The name of a module when loaded if it is in the pre-approved list of modules (see line 128 in src/System.Management.Automation/utils/Telemetry.cs).

Reviewer 3: ```suggestion
- The name of a module when loaded if it is in the pre-approved [list of modules](https://github.com/PowerShell/PowerShell/blob/d74c6bfb7bc2ff84ae15cceeba2fb7dc5c3ccd17/src/System.Management.Automation/utils/Telemetry.cs#L128) 
```

Reviewer 3: If it is not best practice to include this line number perma-link then feel free to ignore.

Modules which are not in this predetermined list are reported as "Anonymous".

> - The type of item being executed (Alias, Function, Filter, Cmdlet, ExternalScript, Application, Script, or Configuration).

Reviewer 2: Just out of curiosity, how useful do we think it is to collect the command type information?

Reviewer 2: @JamesWTruher can you please comment on this question? I'm just curious how useful this information turns out to be in our telemetry analysis so far.

We do not collect the name of the executed item, the parameters used, or values for any parameter.
- When a `PowerShell` object is created
- When a remote session is created

> In addition to the caveats above, the following is _not_ collected:

Reviewer: ```suggestion
In addition to the caveats above, no personally identifiable information is collected.  For example, the following is _not_ collected:
```
